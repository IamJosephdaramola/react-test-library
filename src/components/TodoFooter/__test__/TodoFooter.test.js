import { render, screen, cleanup } from "@testing-library/react";
import TodoFooter from "../TodoFooter";
import { BrowserRouter } from "react-router-dom";

const MockTodoFooter = ({ numberOfIncompleteTasks }) => {
  return (
    <BrowserRouter>
      <TodoFooter numberOfIncompleteTasks={numberOfIncompleteTasks} />
    </BrowserRouter>
  );
};

describe("TodoFooter", () => {
  afterEach(cleanup);

  test("should take a snapshot of the TodoFooter", () => {
    const { asFragment } = render(
      <MockTodoFooter numberOfIncompleteTasks={1} />
    );

    expect(
      asFragment(<MockTodoFooter numberOfIncompleteTasks={1} />)
    ).toMatchSnapshot();
  });

  test("should render tasks when the number of incomplete tasks is five", () => {
    render(<MockTodoFooter numberOfIncompleteTasks={5} />);
    const paragraphElement = screen.getByText(/5 tasks left/i);
    expect(paragraphElement).toBeInTheDocument();
  });

  test("should render tasks when the number of incomplete tasks is one", () => {
    render(<MockTodoFooter numberOfIncompleteTasks={1} />);
    const paragraphElement = screen.getByText(/1 task left/i);
    expect(paragraphElement).toBeInTheDocument();
  });

  test("should link to followers page", () => {
    render(<MockTodoFooter numberOfIncompleteTasks={1} />);
    const linkElement = screen.getByRole("link");
    expect(linkElement).toHaveAttribute("href", "/followers");
  });
});

// test("should render tasks when visible", () => {
//   render(<MockTodoFooter numberOfIncompleteTasks={1} />);
//   const paragraphElement = screen.getByText(/1 task left/i);
//   expect(paragraphElement).toBeVisible();
// });
