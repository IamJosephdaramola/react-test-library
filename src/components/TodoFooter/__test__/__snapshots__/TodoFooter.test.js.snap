// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFooter should take a snapshot of the TodoFooter 1`] = `
<DocumentFragment>
  <div
    class="todo-footer"
  >
    <p>
      1 task left
    </p>
    <a
      href="/followers"
    >
      Followers
    </a>
  </div>
</DocumentFragment>
`;
